aamir_hw2/singly_linkedlist.py

```diff
@@ -0,0 +1,87 @@
+class Node(object):
+	def __init__(self, data = None, next = None):
+		super(Node, self).__init__()
+		self.data = data
+		self.next = next
+
+	def __str__(self):
+		return "{} -> {}".format(self.data, self.next)
+
+class LinkedList(object):
+	def __init__(self):
+		super(LinkedList, self).__init__()
+		self.head = None
+
+	def _get_new_node(self, data):
+		node = Node()
+		node.data = data
+		node.next = None
+
+		return node
+
+	def size(self):
+		count = 0
```

Comment:
The runtime of this implementation would be O(n) so better to keep track of size variable as metadata, which would result in constant O(1) runtime of the impl.

Reply:
This would require a drastic change in the implementation of the class. Cool with that?

Reply:
Ok fine for now! but change it in the doubly linked list.